apisix/core/config_etcd.lua

```diff
@@ -188,9 +191,36 @@ local function run_watch(premature)
 
         ::watch_event::
         while true do
-            local res, err = res_func()
-            if log_level >= NGX_INFO then
-                log.info("res_func: ", inspect(res))
+            local res
+
+            local get_res_th = ngx_thread_spawn(function ()
```

Comment:
The code is fine, but there is a slight impact on performance, because each res_func() will spawn a new thread, and this thread will exit after 50 seconds.
So, is it feasible to run the whole run_watch() in one thread and watch for the exit signal in another thread?

Reply:
Good idea. I have a question, can we use `ngx_thread_spawn` instead of `ngx_timer_at`? we need kill the `run_watch` thread when worker exited.
https://github.com/apache/apisix/blob/c89ec15e85eded367bdc5e03fa61ac5f6e53792d/apisix/core/config_etcd.lua#L269